isobus/src/can_network_manager.cpp

```diff
@@ -524,6 +519,13 @@ namespace isobus
 		}
 	}
 
+	std::shared_ptr<ControlFunction> CANNetworkManager::create_external_control_function(NAME desiredName, std::uint8_t address, std::uint8_t CANPort)
+	{
+		auto controlFunction = std::make_shared<ControlFunction>(desiredName, address, CANPort, ControlFunction::Type::External);
+		controlFunctionTable[CANPort][address] = controlFunction;
```

Comment:
Might be good to add sanity checks to these indexes before we try and use them, just in case we are somehow in an invalid state or in case the way we use this function ever changes, just to be safe